source/common/router/upstream_request.h

```diff
@@ -141,6 +141,7 @@ class UpstreamRequest : public Logger::Loggable<Logger::Id::router>,
   StreamInfo::StreamInfoImpl stream_info_;
   StreamInfo::UpstreamTiming upstream_timing_;
   const MonotonicTime start_time_;
+  absl::optional<uint64_t> response_headers_size_{};
```

Comment:
worth a comment on why we latch original size?

Reply:
Sure thing.